scene/resources/material.cpp

```diff
@@ -353,16 +375,88 @@ Variant ShaderMaterial::get_shader_parameter(const StringName &p_param) const {
 	}
 }
 
+void ShaderMaterial::set_shader_macro_override(const StringName &p_name, const String &p_value) {
+	String value = p_value.strip_edges();
+
+	if (value.is_empty()) {
+		if (macro_cache.erase(p_name)) {
```

Comment:
Do shaders actually support macros that have no value? In C it is common to define one without a value, and the simple fact it is defined is enough to use `#ifdef`. If that is the case in the shading language, then we should not interpret empty string as "erase the macro".

Reply:
Nah, no value intended to remove the macro from the cache, and preprocessor queue.